### PR TITLE
Remove solana-sdk from solana-metrics

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
solana-metrics can compile faster without the solana-sdk dependency

#### Summary of Changes
- Replace solana-sdk with its component crates
- Replace `solana_sdk::clock::Slot` with u64 (Slot is just an alias for u64)

This branches off #3372 so that needs to be merged first (update: done)

### CODE REVIEW

**Comment by yihau** [comment]
thank you for having this. just one small note: I'm not sure about changing `Slot` => `u64` part. to me, it makes the code less clear, even though we all know u64 == Slot. I'd like to defer to @joncinque 's opinion.

**Review by joncinque — COMMENTED** [review]
I tested this with and without `solana-clock`, and the difference was imperceptible in the build time -- can we add `solana-clock` to use `Slot`?

File: perf/Cargo.toml
Comment (joncinque) [inline perf/Cargo.toml:1]: Were the changes here intended?

**Review by joncinque — COMMENTED** [review]
Looks good! Just one last question

File: perf/Cargo.toml
Comment (kevinheavey) [reply perf/Cargo.toml:1]: yes https://github.com/anza-xyz/agave/pull/3374/commits/9bd22da14444e43663a38e0bf326a510e7277ed7

**Review by joncinque — APPROVED** [review]
Ok cool, thanks!

**Pull request merged**